stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,188 @@
+    _stop_batch: int
+    _should_write_train_graph: bool
+    write_steps_per_second: bool
+    _batch_start_time: float
+    _train_step: tf.Variable
+    _should_trace: bool
+    _is_tracing: bool
+    _global_train_batch: int
+    update_freq: int | Literal["epoch"]
+
+    def __init__(
+        self,
+        log_dir: str = "logs",
+        histogram_freq: int = 0,
+        write_graph: bool = True,
+        write_images: bool = False,
+        write_steps_per_second: bool = False,
+        update_freq: int | Literal["epoch"] = "epoch",
+        profile_batch: int | tuple[int, int] = 0,
+        embeddings_freq: int = 0,
+        embeddings_metadata: dict[str, None] | None = None,
+    ) -> None: ...
+    def _write_keras_model_train_graph(self) -> None: ...
+    def _stop_trace(self, batch: int | None = None) -> None: ...
+
+class BackupAndRestore(Callback):
+    def __init__(
+        self, backup_dir: str, save_freq: str = "epoch", delete_checkpoint: bool = True, save_before_preemption: bool = False
+    ) -> None: ...
+
+class RemoteMonitor(Callback):
+    def __init__(
+        self,
+        root: str = "http://localhost:9000",
+        path: str = "/publish/epoch/end/",
+        field: str = "data",
+        headers: dict[Incomplete, Incomplete] | None = None,
+        send_as_json: bool = False,
+    ) -> None: ...
+
+class ReduceLROnPlateau(Callback):
+    def __init__(
+        self,
+        monitor: str = "val_loss",
+        factor: float = 0.1,
+        patience: int = 10,
+        verbose: Literal[0, 1] = 0,
+        mode: Literal["auto", "min", "max"] = "auto",
+        min_delta: float = 1e-4,
+        cooldown: int = 0,
+        min_lr: float = 0,
+        **kwargs: Incomplete,
+    ) -> None: ...
+    def in_cooldown(self) -> bool: ...
+
+class BaseLogger(Callback):
+    def __init__(self, stateful_metrics: Iterable[str] | None = None) -> None: ...
+
+class CSVLogger(Callback):
+    def __init__(self, filename: str, separator: str = ",", append: bool = False) -> None: ...
+
+class EarlyStopping(Callback):
+    monitor_op: Any
+    def __init__(
+        self,
+        monitor: str = "val_loss",
+        min_delta: float = 0,
+        patience: float = 0,
+        verbose: Literal[0, 1] = 0,
+        mode: Literal["auto", "min", "max"] = "auto",
+        baseline: float | None = None,
+        restore_best_weights: bool = False,
+        start_from_epoch: int = 0,
+    ) -> None: ...
+
+class LearningRateScheduler(Callback):
+    def __init__(self, schedule: LearningRateSchedule, verbose: Literal[0, 1] = 0) -> None: ...
+
+class ProgbarLogger(Callback):
+    use_steps: bool
+    def __init__(
+        self, count_mode: Literal["steps", "samples"] = "samples", stateful_metrics: Iterable[str] | None = None
+    ) -> None: ...
+
+class TerminateOnNaN(Callback): ...
+
+class LambdaCallback(Callback):
+    def __init__(
+        self,
+        on_epoch_begin: Callable[[int, Incomplete], Incomplete] | None = None,
```

Comment:
The incomplete for second argument of callable is _Logs. The return type for all the callables is `object` (the return value is ignored.

For the last two callables they should be `Callable[[_Logs], object]`

Reply:
Done in 7a77bd3cc